server/test/controllers/admin/AdminProgramImageControllerTest.java

```diff
@@ -0,0 +1,293 @@
+
+    Result result = controller.index(addCSRFToken(fakeRequest().method("GET")).build(), program.id);
+
+    assertThat(result.status()).isEqualTo(OK);
+    assertThat(contentAsString(result)).contains("fake summary description");
+  }
+
+  @Test
+  public void updateDescription_programNotDraft_throws() {
+    ProgramModel program = ProgramBuilder.newActiveProgram().build();
+
+    assertThatThrownBy(
+            () ->
+                controller.updateDescription(
+                    addCSRFToken(fakeRequest().method("POST")).build(), program.id))
+        .isInstanceOf(NotChangeableException.class);
+  }
+
+  @Test
+  public void updateDescription_missingProgram_throws() {
+    assertThatThrownBy(
+            () ->
+                controller.updateDescription(
+                    addCSRFToken(fakeRequest().method("POST")).build(), Long.MAX_VALUE))
+        .isInstanceOf(NotChangeableException.class);
+  }
+
+  @Test
+  public void updateDescription_createsNewDescription()
+      throws ProgramNotFoundException, TranslationNotFoundException {
+    ProgramModel program = ProgramBuilder.newDraftProgram("test name").build();
+
+    Result result =
+        controller.updateDescription(
+            addCSRFToken(
+                    fakeRequest()
+                        .method("POST")
+                        .bodyForm(ImmutableMap.of("summaryImageDescription", "fake description")))
+                .build(),
+            program.id);
+
+    assertThat(result.status()).isEqualTo(SEE_OTHER);
+    ProgramDefinition updatedProgram = programService.getProgramDefinition(program.id);
+    assertThat(updatedProgram.localizedSummaryImageDescription().isPresent()).isTrue();
+    assertThat(updatedProgram.localizedSummaryImageDescription().get().get(Locale.US))
+        .isEqualTo("fake description");
+  }
+
+  @Test
+  public void updateDescription_editsExistingDescription()
+      throws ProgramNotFoundException, TranslationNotFoundException {
+    ProgramModel program =
+        ProgramBuilder.newDraftProgram("test name")
+            .setLocalizedSummaryImageDescription(
+                LocalizedStrings.of(Locale.US, "first description"))
+            .build();
+
+    Result result =
+        controller.updateDescription(
+            addCSRFToken(
+                    fakeRequest()
+                        .method("POST")
+                        .bodyForm(ImmutableMap.of("summaryImageDescription", "second description")))
+                .build(),
+            program.id);
+
+    assertThat(result.status()).isEqualTo(SEE_OTHER);
+    ProgramDefinition updatedProgram = programService.getProgramDefinition(program.id);
+    assertThat(updatedProgram.localizedSummaryImageDescription().isPresent()).isTrue();
+    assertThat(updatedProgram.localizedSummaryImageDescription().get().get(Locale.US))
+        .isEqualTo("second description");
+  }
+
+  @Test
+  public void updateDescription_editExisting_preservesNonDefaultLocaleTranslations()
```

Comment:
Thanks! I saw similar tests for the statuses work, so also praise to whoever wrote those tests because I was able to re-use a lot of those testing concepts!